realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -325,15 +331,31 @@ public static void setDefaultConfiguration(RealmConfiguration configuration) {
         if (configuration == null) {
             throw new IllegalArgumentException("A non-null RealmConfiguration must be provided");
         }
-        defaultConfiguration = configuration;
+        synchronized (monitorForDefaultConfiguration) {
+            defaultConfiguration = configuration;
+        }
+    }
+
+    /**
+     * Returns the default configuration for {@link #getDefaultInstance()}.
+     *
+     * @return default configuration object or {@code null} if {@link #init(Context)} was not called yet
```

Comment:
Good point. It was just the reference to `Realm.init()` not being called that confused me. I would probably just remove that?